### PR TITLE
MOTOR DRV: Configure microstepping on the fly

### PR DESCRIPTION
**Pull request by StephanCoetzee**
@Jeff-Ciesielski 

Merge https://github.com/keyme/grbl/pull/115 first.

### CODE REVIEW

File: settings.c [hunk 1]
```diff
@@ -244,6 +249,26 @@ uint8_t settings_store_global_setting(int parameter, float value) {
...
+          settings.x_microsteps = value;
+          break;
+        case 48:
+          settings.x_microsteps = value;
+          break;
+        case 49:
+          settings.x_microsteps = value;
+          break;
+      }
+      motor_drv_init();
```
Comment (Jeff-Ciesielski) [inline settings.c]: What would happen if you didn't have SPI enabled?  Seems to me that this doesn't handle both the 3.2 and 4.0 hardware?

File: motor_driver.c [hunk 2]
```diff
@@ -199,22 +202,12 @@ void motor_drv_init()
...
-        steps = QUARTER;
-        break;
-      case 3:
-        steps = SIXTEENTH;
-        break;
-    }
+    uint8_t steps = *(&settings.x_microsteps + idx); 
+
+    char dbg_msg[30];
+    snprintf(dbg_msg, 30, "STEPS: %d", steps);
```
Comment (Jeff-Ciesielski) [inline motor_driver.c]: Might want to remove this debug print or wrap it in a debug flag.

Comment (StephanCoetzee) [reply motor_driver.c]: That wasn't meant to be in the PR. I was just testing it. I'll take it out.

File: settings.c (hunk 1 above)
Comment (StephanCoetzee) [reply settings.c]: Good point. Once https://github.com/keyme/grbl/pull/115 gets merged, I'll rebase and check `settings.spi_motor_drivers` before I call `motor_drv_init()`. Alternatively, I can check for the spi compile flag, but that's going to change once #115 gets merged in any case. Note: as mentioned in https://github.com/keyme/kiosk/pull/10630, the old drivers still rely on the `microsteps` mask, so this shouldn't affect 3.2 kiosks. 

As you mentioned in a comment in https://github.com/keyme/grbl/pull/115, we should expand the motor driver layer to take care of the non-SPI motor drivers as well, but for now, these settings are only for the new SPI motor drivers.

**Comment by Jeff-Ciesielski** [comment]
LGTM once 115 gets in and changes are made re: checking for spi drivers.

**Comment by Jeff-Ciesielski** [comment]
Needs rebase

**Comment by StephanCoetzee** [comment]
Updated and tested. Cuts keys on 3.2 and 4.0.

@Jeff-Ciesielski

**Comment by Jeff-Ciesielski** [comment]
LGTM

**Pull request merged**